### PR TITLE
update helm to install iscsi nfs using daemonset (#2033)

### PR DESCRIPTION
**Pull request by cclhsu**
#### Proposed Changes ####

1. Update helm for daemonset-sa.yaml `longhorn-manager` to install `iscsi` and `nfs` using deamonset/initcontainer.
2. Add flag in values.yaml to enable/disable package host installation.

#### Types of Changes ####

Improvement.

#### Verification ####

1. Deploy RKE/K3S on Centos/Ubuntu/SLES15SP2 and verify if containers run successfully on each node and installed package run successfully.

#### Linked Issues ####

Refs: https://github.com/longhorn/longhorn/issues/2033

#### Further Comments ####

Additional verification is required.

1. Different K8S deployments.
2. End-to-End testing.

NOTE: If initcontainer deployment failed,  `longhorn-manager` will not deploy.

Signed-off-by: cclhsu <clark.hsu@suse.com>

### CODE REVIEW

**Comment by yasker** [comment]
We need to be very careful regarding the automatic installation of those components. Since it might cause the helm install failure, and we need to always make sure the user can recover from the installation/upgrade failure (which is not an easy thing to do with Helm). That's also why I was hesitant to include the init container installation for v1.1.0 release.

For https://github.com/longhorn/longhorn/issues/2033, I consider we're good once we have the NFS client installed as well. Automated installation through Helm is a stretch goal, and I still think it might bring in more trouble than it's worth. 

In short, if we want to do this, we need to figure out all the failure scenarios and the way to recover from them, then decide if it's worth doing it.

**Comment by cclhsu** [comment]
I agreed. I will close the pull request for now.
